cmd/minishift/cmd/root.go

```diff
@@ -203,6 +200,36 @@ func initConfig() {
 	setupViper()
 }
 
+func initializeProfile() string {
+	var (
+		profileName string
+		err         error
+	)
+
+	for i, arg := range os.Args {
+		if arg == "--profile" {
```

Comment:
Why aren't we using `viper.GetString()` to get the value?
ere you are not using the profiile Flag...

Reply:
@gbraad this is because it part of init and till time viper doesn't have all required data in.

Reply:
@praveenkumar but it does not use https://github.com/minishift/minishift/pull/1380#discussion-diff-138819260R54 to construct the flag, which makes tracking the usage very difficult and error-prone. Currently already happening. I suggest https://github.com/minishift/minishift/pull/1380#discussion-diff-138819260R54 to be named `profileFlag` or `profileNameFlag` and construct it here as:

```go
        if arg = "--" + profileFlag {
```

Reply:
@gbraad I do like this approach 👍

Reply:
Yup. I agree.